### DISCUSSION TITLE
List external HTTP referrers and incoming links to issues and PRs

### DISCUSSION THREAD

**Discussion by daiplusplus** [community/community#10290] [Projects and Issues]
Presently GitHub Issues' timelines will show when an issue or pull-request is referenced-by or linked-to from another GitHub issue or PR, which is handy, for example:

![image](https://user-images.githubusercontent.com/1693078/150078521-66785908-bc97-4590-bb14-90b3c77524a1.png)

However I'd like to see the same for public external websites that link to specific GitHub issues and PRs, because that can be a source of a lot of useful information, especially StackOverflow QAs and links from other source-control and project-management systems (though these aren't necessarily public...).

I'd like to see 2 things, really:

* When any incoming HTTP request for a GitHub Issue or Pull-Request has a `Referer` (sic) header back to a public web-page, then...:
  * Make an anonymous GET request for the `Referer` page and verify that the page does actually does contain a link-back. This ensures only _public_ pages will appear. And re-verify every stored link once every 30 days or so?
  * Restrict it to known websites only to mitigate spam (honestly I'd be happy with just StackOverflow and a few other sites), otherwise it'll be a repeat of [the Trackback Spam](https://theory.stanford.edu/people/jcm/papers/blogspam_CCSW09.pdf) we had in the 00s.
* Continuously spider the public web to search for incoming links to GitHub Issues and PRs even if there's no logged `Referer` headers.
  * This will likely be the only way to implement this in the near-future as the `Referer` header will probably be gone within a decade from now (given its negative impact on user's privacy, and increasingly tighter browser and website CSP policies).

**Comment by corneliusroemer** [comment]
This would be amazing. In particular, I'd love this feature for links from Github discussions to PRs/issues/commits.
Until this is implemented, maybe someone can make a bot that does this?
